test/unit/mock-tooling.ts

```diff
@@ -19,4 +30,26 @@ describe('mock-tooling', () => {
       return expect(() => mockTooling.mockTokeninfoEndpoint({ url: ''} as MockOptions)).to.throw(Error);
     });
   });
+
+  describe('mockAccessTokenEndpointWithErrorResponse', () => {
+    it('should throw error, if url is empty', () => {
+      return expect(() => mockTooling.mockAccessTokenEndpointWithErrorResponse({ url: ''} as MockOptions, HttpStatus.BAD_REQUEST)).to.throw(Error);
+    });
+
+    it('should return specified/configured status code', () => {
+      const url = 'https://www.github.com';
+      const expectedState = HttpStatus.BAD_REQUEST;
+      mockTooling.mockAccessTokenEndpointWithErrorResponse({ url } as MockOptions, expectedState);
+
+      const result = fetch(url, { method: 'POST' })
+        .then((data) => data.status);
+      return expect(result).to.become(expectedState);
+    });
+  });
+
+  describe('mockTokeninfoEndpointWithErrorResponse', () => {
```

Comment:
Maybe we could just test `mockEndpointWithErrorResponse` so we do not have to write each test multiple times?

Reply:
I am not sure if the test coverage would be lowered in this case...